rita_common/src/usage_tracker/mod.rs

```diff
@@ -133,44 +162,94 @@ impl UsageTracker {
-                                blank_usage_tracker
-                            }
-                        }
-                    }
+        let file_exists =
+            Path::new(&settings::get_rita_common().network.usage_tracker_file).exists();
+        let file_is_bincode = Path::new(&settings::get_rita_common().network.usage_tracker_file)
+            .extension()
+            .unwrap()
+            .eq("bincode");
+        fn decompressed(mut file: File) -> Result<Vec<u8>, RitaCommonError> {
+            let mut byte_contents = Vec::new();
+            match file.read_to_end(&mut byte_contents) {
+                Ok(_bytes_read) => {
+                    let mut decoder = ZlibDecoder::new(&byte_contents[..]);
+                    let mut bytes = Vec::<u8>::new();
+                    decoder.read_to_end(&mut bytes)?;
+
+                    Ok(bytes)
+                }
+                Err(e) => {
+                    error!("Failed to read usage tracker file! {:?}", e);
+                    Err(RitaCommonError::StdError(e))
+                }
+            }
+        }
+        let file = File::open(settings::get_rita_common().network.usage_tracker_file);
+        let fileopen = match file {
+            Ok(file) => file,
+            Err(e) => {
+                error!("Failed to open usage tracker file! {:?}", e);
+                return blank_usage_tracker;
+            }
+        };
+        match (file_exists, file_is_bincode, decompressed(fileopen)) {
+            (true, true, Ok(bytes)) => {
+                //bincode deserialization
+                let deserialized: Result<UsageTracker, Error> = bincode::deserialize(&bytes);
+                match deserialized {
+                    Ok(usage) => usage,
                     Err(e) => {
-                        error!("Failed to read usage tracker file! {:?}", e);
+                        error!(
+                            "Failed to deserialize bytes in compressed bw history {:?}",
+                            e
+                        );
                         blank_usage_tracker
                     }
                 }
             }
-            Err(e) => {
-                error!("Failed to open usage tracker file! {:?}", e);
-                blank_usage_tracker
+            (true, false, Ok(bytes)) => {
+                //serde deserialization
+                let deserialized: Result<UsageTracker, SerdeError> = serde_json::from_slice(&bytes);
+                let mut res = match deserialized {
+                    Ok(usage) => usage,
+                    Err(e) => {
+                        error!(
+                            "Failed to deserialize bytes in compressed bw history {:?}",
+                            e
+                        );
+                        blank_usage_tracker
+                    }
+                };
+                let mut rita_common = get_rita_common();
+                rita_common.network.usage_tracker_file =
```

Comment:
comments here about what's being done, we're changing the file name

also we should change only the filename on the end of the string, what if the user had provided /my/strange/filepath/tracker.json, we just want to take everything before the dot (their filename) and append .bincode as the new suffix

Reply:
fixed